File: 3255-find-the-power-of-k-size-subarrays-ii/3255-find-the-power-of-k-size-subarrays-ii.cpp
class Solution {
public:
    vector<int> resultsArray(vector<int>& nums, int k){
        int n=nums.size();
        vector<int> bk(n,0);
        for(int i=1;i<n;i++){
            if(nums[i-1]+1==nums[i]){bk[i]=1;}
            bk[i]+=bk[i-1];
        }
        vector<int> res;
        for(int i=k-1;i<n;i++){
            int jud=bk[i]-bk[i-k+1];
            if(jud==(k-1)){res.push_back(nums[i]);}
            else{res.push_back(-1);}
        }
        return res;
    }
};

File: 3255-find-the-power-of-k-size-subarrays-ii/3255-find-the-power-of-k-size-subarrays-ii.java
class Solution {
    public int[] resultsArray(int[] nums, int k) {
        int n = nums.length;
        int[] ct = new int[n];
        ct[0] = 1;
        for (int i = 1; i < n; ++i) {
            if (nums[i]==nums[i-1]+1) {
                ct[i] = ct[i-1]+1;
            } else {
                ct[i] = 1;
            }
        }
        int[] ans = new int[n-k+1];
        for (int i = 0; i < n-k+1; ++i) {
            if (ct[i+k-1]>=k) {
                ans[i] = nums[i+k-1];
            } else {
                ans[i] = -1;
            }
        }
        return ans;
    }
}